Open Addressing Hash Table (Constructors)

Added All constructors for Open Addressing Hash Table

BlackCorbeau opened the pull request:
Please mark my lib_HashTable. I'm not sure that im thinking on right way.

RemizovKL committed:
fix some bugs

RemizovKL committed:
Functionalization of Queue and list have been changed a bit, and now they work how thay need to work

RemizovKL committed:
All work with iterators have been done, and also had some fixes with all last work

RemizovKL committed:
Added a RabitandTurtle function and get a bit of changes

RemizovKL committed:
Fix commit for mege

RemizovKL committed:
added class DSU

RemizovKL committed:
Added a based form of DSU

RemizovKL committed:
Added a based form of DSU part 2

RemizovKL committed:
Added optimization

RemizovKL committed:
Added tests

RemizovKL committed:
Added interface

Added ITable interface for realesed another tables

RemizovKL committed:
Working with CString

Added function with is regenerate int to CString object

RemizovKL committed:
Worcing with PArser libruary

Added fucktion with is can generate random int Codes for UTable

RemizovKL committed:
Working with lib_List

Added a fucntion witch can send size of List

RemizovKL committed:
Added Unsorted Table Libruary

Added a class with all methods annd also added a constructors and insert
method

RemizovKL committed:
UTable update

Added realization of functions eraise and find

RemizovKL committed:
UtableDM

Added Unsorted table with is made on DMassiv, and also all test have
been wroted

RemizovKL committed:
STable

Added an ordered tabel on DMassiv, witn tests

RemizovKL committed:
BinTree

Starts work with binary tree

RemizovKL committed:
Fix of last Libs

Had some thikses with class TPair and also sdded an operator[] for
DMassiv

RemizovKL committed:
added some fixses for TPair

RemizovKL committed:
BinTree Table

Initialized a bin tree table class

RemizovKL committed:
Bin Tree Table (Insert out of key)

Added insert method for with auto key

RemizovKL committed:
Bin Tree Table (Insert With Key)

Added insert method with key

RemizovKL committed:
Bin Tree Table (Eraise)

Added eraise method to delete by value (don`t work)

RemizovKL committed:
Bin Tree Table (Find)

Added a find method to find by value

RemizovKL committed:
Bin Tree Table (Print)

Added a little method to print table

RemizovKL committed:
Bin Tree Table (SizeGetterTest)

added a little size test

RemizovKL committed:
Heap (Initialization)

Added initialization of class Heap

RemizovKL committed:
Heap (Constructors)

Added constructors for Heap

RemizovKL committed:
Heap (subFunctional)

Added methods:
parent which is returned position of parent element in DMassiv
right witch is returned position of right child element in DMassiv
left witch is returned position of left chidl element in DMassiv

RemizovKL committed:
Refactoring

Added a hronological order in main CMakeFiles.txt

RemizovKL committed:
Heap (find)

Added method find for heap ( reason for this is convenience of testing)

RemizovKL committed:
Heap (isEmpty)

Added method is empty for checkoing is Heap empty

RemizovKL committed:
Tastes revork

RemizovKL committed:
Heap (Insert)

Added method Insert for heap

RemizovKL committed:
Heap (eraise)

Added Method Eraise for Heap

RemizovKL committed:
Heap (Tests)

Added tests for heap:
DefaultConstructor; TypeConstructor; ArrayConstructor; InsertAndShiftUp;
EraseAndShiftDown; HeapPropertiesMax; HeapPropertiesMin;
FindNonExistent; EmptyHeapOperations; LargeInput

RemizovKL committed:
SubTask for heap

Subtask for Heap to sorting massiv by heap is clear and can be shown

RemizovKL committed:
Heap (rework)

Heap has been reworked with pointer to funktion of min max or user

RemizovKL committed:
HashNode (Initialized)

Added a a special data type for Open Addressing Hash Table

RemizovKL committed:
Open Addressing Hash Table (HashFunktion)

Added hash function for Open Addressing Hash Table ?

RemizovKL committed:
Open Addressing Hash Table (Initialization)

Initialized Open Addressing Hash Table class

RemizovKL committed:
Open Addressing Hash Table (Constructors)

Added All constructors for Open Addressing Hash Table

NiTro005 commented on lib_HashTable/HashTable.h:
Просто вот скажи, зачем тебе здесь высчитывать Capacity если ты можешь просто в define определить константный размер массива, а пользователя вообще ограничить от этого.

NiTro005 commented on lib_HashTable/HashTable.h:
что это и для чего

NiTro005 reviewed (commented):
Так-то нормально только вот это странности, хотя бы объясни для чего это все.

BlackCorbeau commented on lib_HashTable/HashTable.h:
А почему нет, да, я понимаю что хэш таблица это стандартная структура данных, но она может потребоваться под определенный размер изначально.

BlackCorbeau commented on lib_HashTable/HashTable.h:
Это ссылка на функцию хэширования,  она нужна что бы в случае чего пользователь мог написать свою функцию хэширования и таблица работала с ней.

NiTro005 commented on lib_HashTable/HashTable.h:
обычно специфические хэш таблицы отдельно реализовывают, а стандарт он статический , но я влюбом случае тогда не понимаю резон define 15 вот ты решил создать массив на лям элементов, и что тебе с этого define , будет миллион и 15 элементов

NiTro005 commented on lib_HashTable/HashTable.h:
хорошо ) но мило как ты узнал новую фишку и стал ей везде пользоваться

NiTro005 commented on lib_HashTable/HashTable.h:
посмотрим докопается она или нет до стдшного хэша

BlackCorbeau commented on lib_HashTable/HashTable.h:
Да, ты абсолютно прав

BlackCorbeau commented on lib_HashTable/HashTable.h:
Я как раз таки не узнал, я решил ею больше пользоваться что бы лучше понять

RemizovKL committed:
Open Adress Hash Table (Insert)

Added insert method for oprn adress Hash Table

RemizovKL committed:
refactor(OpenAdressHashTable): Renamed file for class OAHashT

Do it for reason that its better for including and moduling

RemizovKL committed:
fix(OpenAdressHashTable): Fix HashFunc

HashFunc has been divided into functions for hashing and funktion for probe with is protected colisees

RemizovKL committed:
fix(OpenAdressHashTable): Rewrite Insert with using probe

RemizovKL committed:
feat(OpenAdressHashTable): added print method

Added Print method for debugging better

RemizovKL committed:
feat(OpenAdressHashTable): Added find method

RemizovKL committed:
feat(OpenAdressHashTable): Added eraise method

NiTro005 commented on lib_HashTable/OpenAdressHashTable.h:
Тут единственный совет сделать эту структуру внутри класса, тогда не нужны будут геттеры и сеттеры, а пользователь не сможет создать экземляр этой структуры(ему и не нужно)

NiTro005 commented on lib_HashTable/OpenAdressHashTable.h:
если это пользовательский метод, то зачем возвращать ноду, пользователю нужно только значение

NiTro005 reviewed (commented):
Ну я нашел только 1 плохой момент, у тебя везде дублирование кода +- одинаковое, поиск свободного хэша для вставки и поиск хэша для удаления лучше вывести в отдельные функции  . Тоесть сделать служебные функции поиска, а пользовательский find пусть вызывает что нужно + добавь перегрузку квадратных скобок для удобства, а так все хорошо.

RemizovKL committed:
test(OpenAdressHashTable): Added tests for open adress hash table

RemizovKL committed:
feat(ListHashTable): structure ListNode

Initialized class ListNode with will be a main usage class in List Hash Table.
Added:
  Based constructor;
  Initialized constructor;
  Setters and getters for list with values;
  Setters and getters for state;

RemizovKL committed:
feat(ListHashTable): Class Iniitializaton

Initialized class of List Hash Table (only based fields and methods)

RemizovKL committed:
feat(ListHashTable): Constructors

Added based, initialization, and partial initialization constructors for List Hash Table

RemizovKL committed:
feat(listHashTable): HashFunc

RemizovKL committed:
fix(ListHashTable): ListNode

Deleted struct ListNode because it was useless

RemizovKL committed:
fix(ListHashTable): rebuild

rebuild of fields and methods besause of deleting HashNode

RemizovKL committed:
fix(ListHashTable):insert with ListNode

Adde insert with list node because it can be using

RemizovKL committed:
feat(ListHashTable): realization insert for list

Added realization for insert with list

RemizovKL committed:
feat(ListHashTable): realization for insert with node

added realization for insert method with node with is in ./lib_list/list.h

RemizovKL committed:
fix(ListHashTable): reinitialization of class

Reinitialize class of List Hash Table because of production necessity.

RemizovKL committed:
fix(ListHashTable): rewriting Insert

Fix problem with paste into empty massiv

RemizovKL committed:
feat(listHashTable): find

RemizovKL committed:
feat(ListHashTable): eraise
